### DISCUSSION TITLE
unresponsive

### DISCUSSION THREAD

**Discussion by oddball** [graphql/graphiql#3186] [Ideas]
I have a query that returns an array of dicts. Perhaps a 150 000 rows. Unzipped it is 32 MB, zipped 2.6 MB. 

The query is quick, but GraphiQL shows this message

<img width="383" alt="unresponsive" src="https://github.com/graphql/graphiql/assets/1045794/f8620f32-ca97-4ed7-9d61-6de25739608c">

The thing that takes time (I think), to calculate is the folding "button" encircled in the picture below. 

<img width="174" alt="folding_button" src="https://github.com/graphql/graphiql/assets/1045794/9fd00be4-7427-4450-b258-6d064f24d654">

Would it be possible not to generate the folding buttons when data is bigger than X ?

**Comment by acao** [comment]
@oddball good question! disabling this functionality internally is not hard. worth considering a "lightweight" editor mode
